MAP-13: MetadataTerm.metadataClass and MetadataTerm.metadataUuid shou…

jvena1 opened the pull request:
ref: https://issues.openmrs.org/browse/MAP-13

suniala commented:
Please notice: "This branch has conflicts that must be resolved"

A rebase on openmrs:master is needed. @jvena1, are you familiar with the process of rebasing and resolving conflicts? If not, I can give you some pointers.

suniala commented on omod/src/main/resources/liquibase.xml:
```diff
@@ -195,5 +195,31 @@
+						  columnName="metadata_class" />
+		</preConditions>
+		<comment>
+			Make MetadataTerm.metadataClass optional
```
If you want to refer to the class name here, it is actually MetadataTermMapping.

jvena1 replied:
Here we refer to the table name (rather than the class name) in the hibernate mapping, please see [MetadataTerm.hbm.xml](https://github.com/openmrs/openmrs-module-metadatamapping/blob/df90370adde14251bdef0538c8f3ddb8b0fd2544/api/src/main/resources/MetadataTerm.hbm.xml#L19).

suniala commented on omod/src/main/resources/liquibase.xml:
```diff
@@ -195,5 +195,31 @@
-
+	
+	<changeSet id="metadatamapping-2016-01-06-0800" author="jasonvena">
+		<preConditions onFail="MARK_RAN">
```
I suggest you add a sort of "disclaimer" comment here. Something like: "Liquibase does not have a precondition for checking 'not null'. Instead, we follow the openmrs convention of just checking if the column itself exists."

jvena1 replied:
will do, I'll just shorten it a wee bit :)

suniala commented on omod/src/main/resources/liquibase.xml:
```diff
@@ -195,5 +195,31 @@
+	</changeSet>
+	
+	<changeSet id="metadatamapping-2016-01-06-0801" author="jasonvena">
+		<preConditions onFail="MARK_RAN">
```
The same comment here.

suniala commented on omod/src/main/resources/liquibase.xml:
```diff
@@ -195,5 +195,31 @@
+						  columnName="metadata_uuid" />
+		</preConditions>
+		<comment>
+			Make MetadataTerm.metadataUuid optional
```
Same note about MetadataTermMapping.

jvena1 replied:
Same as above, we refer to the table name (rather than the class name) in the hibernate mapping, please see [MetadataTerm.hbm.xml](https://github.com/openmrs/openmrs-module-metadatamapping/blob/df90370adde14251bdef0538c8f3ddb8b0fd2544/api/src/main/resources/MetadataTerm.hbm.xml#L19).

jvena1 commented:
I sure am... There were zero conflicts when I rebased and then submitted the PR, but I see that another commit was made today before this one. I'll rebase again after addressing the comments above, thanks!
